kubespawner/objects.py

```diff
@@ -566,3 +596,107 @@ def make_ingress(
     )
 
     return endpoint, service, ingress
+
+
+def make_secret(
+    name,
+    username,
+    ssl_directory,
+    labels=None,
+    annotations=None,
+):
+    """
+    Make a k8s secret specification for mount ssl credentials for running a user notebook.
+
+    Parameters
+    ----------
+    name:
+        Name of the secret. Must be unique within the namespace the object is
+        going to be created in.
+    username:
+        The name of the user notebook.
+    ssl_directory:
+        Path to the SSL directory for certificates and keys.
```

Comment:
I would like to add something about this directory's structure, either by declaring it to look like ...

        /user-{username}/user-{username}.key
        /user-{username}/user-{username}.crt
        /notebooks-ca/notebooks-ca.crt
        /hub-ca_trust.crt

... or by referencing it is created by JupyterHub automatically or something like that.

```suggestion
        Path to a directory containing all users certificates and keys.
```

Reply:
I'm good either way what do you prefer?  I can link out to the JupyterHub docs.  I can inline your above example of the directory structure.

Reply:
I don't yet understand this directory, is jupyterhub generating these certs? Referencing that JupyterHub is meant to generate it all, and the docs about that would be the best option in my mind.

Reply:
JupyterHub can create it and create all of the certs or the user can also do it.

Reply:
Let me know if this makes sense
```
        Location of JupyterHub's internal_ssl directory.  This directory is managed by JupyterHub itself.  
        See https://jupyterhub.readthedocs.io/en/stable/api/app.html
```

Reply:
I can provide a suggestion if we get this right first.

https://github.com/jupyterhub/kubespawner/pull/386#pullrequestreview-386026000